cmake/ProtoBuf.cmake

```diff
@@ -41,10 +38,14 @@ macro(custom_protobuf_find)
     endif()
   endif()
 
+  set(__caffe2_CMAKE_POSITION_INDEPENDENT_CODE ${CMAKE_POSITION_INDEPENDENT_CODE})
```

Comment:
If CAFFE2_LINK_LOCAL_PROTOBUF is ON, then won't this line override the original CMAKE_POSITION_INDEPENDENT_CODE that was saved into __caffe2_CMAKE_pic ?

Reply:
I guess you are saying __caffe2_CMAKE_POSITION_INDEPENDENT_CODE, these variables are meant to be tmp.

Reply:
if CAFFE2_LINK_LOCAL_PROTOBUF is ON then line 33 will overwrite CMAKE_POSITION_INDEPENDENT_CODE to ON, and then line 41 will always set __caffe2_CMAKE_POSITION_INDEPENDENT_CODE to ON, then line 46 will always set CMAKE_POSITION_INDEPENDENT_CODE back to ON

Reply:
discussed with @pjh5 offline, there is no issue.